ui/src/components/layout/FlowCanvas.tsx

```diff
@@ -102,15 +105,19 @@ const FlowCanvas: React.FC<Props> = () => {
    */
   const onInit = async (reactFlowInstance: ReactFlowInstance) => {
     setReactFlowInstance(reactFlowInstance);
-
-    const store = getDataStore();
-    const planId = store.getCursor('root.planId');
-    const displayNodes: DisplayNode[] = store.getDisplayNodes(planId);
-    const displayEdges: DisplayEdge[] = store.getDisplayEdges(planId);
-    setNodes(displayNodes.map((node: DisplayNode) => displayNodeToNode(node)));
-    setEdges(displayEdges.map((edge: DisplayEdge) => displayEdgeToEdge(edge)));
   };
 
+  /**
+   * These use effect calls are not firing when there's a remote update. I'm not sure why.
+   */
+  useEffect(() => {
```

Comment:
Even when these do get to firing we need to use some of the internals to wire up the ReactFlows internal Zustand store actions: https://reactflow.dev/docs/api/hooks/use-store/#internal-actions

Reply:
So, some context is probably helpful. Even if it's just me coming back to this in a few hours. When we update a node in the canvas, the store gets the change, dispatches to the subscribers (react), it sends out the signal, the local screen updates. On the other side, it receives the signal, it updates the tree, it dispatches the subscribers, but then somehow it doesn't detect the difference, so the useEvent never fires on the clients receiving the signal.

Reply:
You can even log the snapshots and see that it changed between the two snapshots. So I really don't know why the useEffect isn't firing.